airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/util/FastUUIDGenerator.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.util
+
+import com.github.f4b6a3.uuid.alt.GUID
+import jakarta.inject.Singleton
+import java.nio.ByteBuffer
+import java.util.UUID
+import java.util.concurrent.ThreadLocalRandom
+
+/**
+ * Custom UUID generator to avoid locking caused by the use of [java.security.SecureRandom] by the
+ * JDK [UUID] implementation. In testing, this implementation, while not cryptographically secure,
+ * outperforms the secure implementation by about 4x. <p /> <p /> <b>N.B.</b> This generator should
+ * only be used when a cryptographically secure UUID is not necessary.
+ */
+@Singleton
+class FastUUIDGenerator {
+
+    /**
+     * Generates a random UUID v4 random-based UUID with Unix epoch. This method is up to 10 times
+     * faster than [UUID.randomUUID].
+     */
+    fun randomUUID(): UUID = GUID.v4().toUUID()
```

Comment:
I'd name this v4

Reply:
or add a v7 and use that

Reply:
https://github.com/airbytehq/airbyte/pull/61528/files#diff-c202da441aa7cbdc0cd580d9a9b4b52fa5bc6da6f33752c2415a89b44d986b73R8

Reply:
Done